source3/modules/vfs_zfsacl.c

```diff
@@ -36,13 +36,132 @@
 
 #define ZFSACL_MODULE_NAME "zfsacl"
 
+static struct SMB4ACL_T *zfsacl_defaultacl(TALLOC_CTX *mem_ctx, 
+					SMB_STRUCT_STAT *psbuf)
+{
+       struct SMB4ACL_T *pacl = NULL;
+       struct SMB4ACE_T *pace;
+	mode_t mode = psbuf->st_ex_mode;
+	uint32_t modify_set = SMB_ACE4_READ_DATA | SMB_ACE4_READ_ACL \
+		| SMB_ACE4_WRITE_DATA | SMB_ACE4_APPEND_DATA | SMB_ACE4_READ_NAMED_ATTRS \
+		| SMB_ACE4_WRITE_NAMED_ATTRS | SMB_ACE4_EXECUTE | SMB_ACE4_DELETE_CHILD \
+		| SMB_ACE4_READ_ATTRIBUTES | SMB_ACE4_WRITE_ATTRIBUTES | SMB_ACE4_DELETE \
+		| SMB_ACE4_SYNCHRONIZE;
+	uint32_t read_set = SMB_ACE4_READ_DATA | SMB_ACE4_READ_ACL \
+		| SMB_ACE4_READ_NAMED_ATTRS | SMB_ACE4_EXECUTE | SMB_ACE4_READ_ATTRIBUTES \
+		| SMB_ACE4_SYNCHRONIZE;
+	uint32_t write_only_set = SMB_ACE4_READ_ACL | SMB_ACE4_WRITE_DATA \
+		| SMB_ACE4_READ_NAMED_ATTRS | SMB_ACE4_WRITE_NAMED_ATTRS | SMB_ACE4_EXECUTE \
+		| SMB_ACE4_READ_ATTRIBUTES | SMB_ACE4_WRITE_ATTRIBUTES | SMB_ACE4_DELETE \
+		| SMB_ACE4_SYNCHRONIZE;
+	uint32_t base_set = SMB_ACE4_READ_NAMED_ATTRS | SMB_ACE4_READ_ATTRIBUTES \
+		| SMB_ACE4_READ_ACL;
+
```

Comment:
I'd move those mode definitions outside of the function body. Possibly make macro definitions, like:
```
#define ZFSACL_MODIFY_SET (SMB_ACE4_READ_DATA | SMB_ACE4_READ_ACL \ 
		| SMB_ACE4_WRITE_DATA | SMB_ACE4_APPEND_DATA | SMB_ACE4_READ_NAMED_ATTRS \ 
		| SMB_ACE4_WRITE_NAMED_ATTRS | SMB_ACE4_EXECUTE | SMB_ACE4_DELETE_CHILD \ 
		| SMB_ACE4_READ_ATTRIBUTES | SMB_ACE4_WRITE_ATTRIBUTES | SMB_ACE4_DELETE \ 
		| SMB_ACE4_SYNCHRONIZE)
```